src/core/CorePool.ts

```diff
@@ -113,10 +127,31 @@ export class CorePool {
     tickUpper: number,
     amount: JSBI
   ): { amount0: JSBI; amount1: JSBI } {
-    // TODO
+    let amount0: JSBI;
+    let amount1: JSBI;
+
+    ({ amount0: amount0, amount1: amount1 } = this.modifyPosition(
+      owner,
+      tickLower,
+      tickUpper,
+      amount
+    ));
```

Comment:
let { position, amount0, amount1 } = this.modifyPosition(
      owner,
      tickLower,
      tickUpper,
      JSBI.unaryMinus(amount)
    );